code/game/machinery/alarm.dm

```diff
@@ -129,16 +133,8 @@
 	unregister_radio(src, frequency)
 	return ..()
 
-/obj/machinery/alarm/Initialize(mapload, var/dir, atom/frame)
-	. = ..(mapload, dir)
-
-	if(istype(frame))
-		buildstage = 0
-		wiresexposed = 1
-		pixel_x = (dir & 3)? 0 : (dir == 4 ? -21 : 21)
```

Comment:
wouldn't removing this hackery break the mapped alarms offsets?

Reply:
It's also set in `update_icon`.